Updated Android low-level settings guide (#185)
AdguardTeam/KnowledgeBase: i18n/be/docusaurus-plugin-content-docs/current/adguard-for-android/solving-problems/low-level-settings.md

> Changing *Low-level settings* can cause problems with the performance of AdGuard, may break the Internet connection or compromise your security and privacy. You should only open this section if you are sure of what you are doing or our support-team has asked you about it.

> To go to *Low-level settings*, open AdGUard and tap the gear icon on the bottom-right corner of the screen. Then choose *General → Advanced → Low-level settings*.

Reviewer: AdGUard

Reviewer: on the bottom-right corner —> in the lower right corner

## Low-level settings

> For Android v4.0 we completely redesigned the low-level settings: divided them into logic blocks, made them clearer, added validation of entered values and other safety valves, got rid of some and added some other settings. 

Reviewer: можно подумать, что это версия андроида, лучше For AdGuard v4.0 for Android

Reviewer: ```suggestion
For AdGuard v4.0 for Android we've completely redesigned the low-level settings: divided them into thematic blocks, made them clearer, added validation of entered values and other safety valves, got rid of some settings, and added others. 
```

### DNS protection

#### Fallback upstreams 

> Here you can specify the fallback DNS resolver(s) to be used if the configured server is unavailable. Three options exist: *Automatic DNS*, *None*, and *Custom DNS*. If no fallback server is specified, the *Automatic DNS* - the system DNS or AdGuard DNS - will be used. A string of *None* means no fallback at all. By selecting *Custom DNS*, you can list IPv4 and IPv6 server addresses to use as upstreams.

Reviewer: ```suggestion
Here you can specify the fallback DNS resolver(s) to be used if the configured server is unavailable. There are three options: *Automatic DNS*, *None*, and *Custom DNS*. If no fallback server is specified, the *Automatic DNS* — the system DNS or AdGuard DNS — will be used. *None* means no fallback at all. Selecting *Custom DNS* allows you to list IPv4 and IPv6 server addresses to use as upstreams.
```

#### Fallback domains

> Here you can list domains that will be directly forwarded to the fallback upstreams if they exist.

Reviewer: ```suggestion
Here you can list domains that will be forwarded directly to fallback upstreams if they exist.
```

#### Detect search domains

> If enabled AdGuard detects search domains and automatically forwards them to the fallback upstreams if they exist.

Reviewer: ```suggestion
If this option is enabled, AdGuard will detect search domains and automatically forward them to fallback upstreams.
```

Reviewer: а, ну или в формате как ниже:
If enabled, AdGuard strips Encrypted Client Hello parameters from DNS responses.

#### Bootstrap upstreams

Bootstrap DNS for DoH, DoT, and DoQ servers. The *Automatic DNS* - the system DNS or AdGuard DNS - is used by default. By selecting *Custom DNS*, you can list IPv4 and IPv6 server addresses to use as bootstrap upstreams.

#### Blocking mode for adblock-style rules

Here you can specify the response type for domains blocked by DNS rules based on adblock rule syntax (for instance, `||example.org^`).

> 

Reviewer: лишняя пустая строка

#### Validate DNS upstreams

Enable to make AdGuard test DNS upstreams before adding or updating custom DNS servers.

### Filtering

#### Capture HAR

Here you can enable HAR file capture. Use it only for debugging purposes! If the setting is enabled, AdGuard will create a directory named "har" inside the app cache directory. It contains information about all filtered HTTP requests in HAR 1.2 format and can be analyzed with the Fiddler program.

### HTTPS filtering

#### Encrypted Client Hello

Every encrypted Internet connection has an unencrypted part. This is the very first packet which contains the name of the server you are connecting to. Encrypted Client Hello technology is supposed to solve this issue and encrypt that last bit of unencrypted information. To benefit from it, enable the *Encrypted Client Hello* option. It uses a local DNS proxy to look for ECH configuration for the domain. If it is found, ClientHello packet will be encrypted.

#### OCSP checking

Once enabled, this option runs asynchronous OCSP checks to check whether the website’s SSL certificate is revoked. 

> If the OCSP check completes within the minimum timeout, AdGuard will immediately apply the result: block the connection if the certificate is revoked or establish a connection if the certificate is valid. 

Reviewer: ```suggestion
If the OCSP check is completed within the minimum timeout, AdGuard will immediately block the connection if the certificate is revoked or establish the connection if the certificate is valid. 
```

> If the verification takes too long, AdGuard will establish a connection and continue checking in the background. If the certificate is revoked, current and future connections to the domain will be blocked.

Reviewer: ```suggestion
If the verification takes too long, AdGuard will establish a connection and continue checking the certificate in the background. If it is revoked, current and future connections to the domain will be blocked.
```

#### Redirect DNS-over-HTTPS requests

IF enabled, DNS-over-HTTPS requests will be redirected to the DNS Protection module. We recommend disabling fallback upstreams and use only encrypted DNS servers to maintain privacy.

> ### Outbound Proxy

Reviewer: proxy

Author: В приложении написано с большой буквы

#### Show the Filter DNS requests setting

When this feature is enabled, in the *Settings ➝ Filtering ➝ Network ➝ Proxy ➝ Proxy server ➝ Add proxy server* section, the string *Filter DNS requests* appears with the switch opposite it. By toggling the switch, you can enable filtering of DNS requests that pass through the proxy. 

### Protection

#### Port ranges

Here you can specify port ranges that should be filtered. 

#### Log removed HTML events

If enabled, AdGuard writes info about blocked HTML elements to the filtering log.

#### Scriplet debugging

If you need to activate debugging of scriptlets, enable this feature. Then there will be messages in the browser log that some scriplet rules have been applied.

#### Excluded apps

> Here can be listed package names and UIDs you want to exclude from AdGuard protection.

Reviewer: ```suggestion
Here you can list package names and UIDs that you want to exclude from AdGuard protection.
```

> #### QUIC bypass packages

Reviewer: не packets?

Author: нет

Here you can specify package names for which AdGuard should bypass QUIC traffic.

#### IPv6 filtering

If enabled, AdGuard filters IPv6 networks if an IPv6 network interface is available.

#### IPv4 ranges excluded from filtering 

Filtering for IPv4 ranges, listed in this section, will be disabled.

#### IPv6 ranges excluded from filtering 

Filtering for IPv6 ranges, listed in this section, will be disabled.

#### TCP keepalive for outgoing sockets

If enabled, AdGuard sends a keepalive probe after the specified time period to ensure if the TCP connection is alive. Here you can specify the TCP keepalive idle time before starting keepalive probes and time between keepalive probes for an unresponsive peer.

> After a system defined number of unsuccessful attempts to get a response from the server, the system automatically closes the TCP connection.

Reviewer: system-defined number

### Local VPN settings

#### Recovery delay for revoked VPN

Here you can set the time of a delay in milliseconds before AdGuard will try to restore VPN protection after it has been revoked by a third-party VPN app or by deleting the VPN profile. The default value is 5000 ms.

#### Reschedule delay for revoked VPN recovery

Here you can set the time of a delay in milliseconds before AdGuard will reschedule the restoration of VPN protection after it has been revoked by a third-party VPN app or by deleting the VPN profile. The default value is 5000 ms.

#### MTU

Here you can set the maximum transmission unit (MTU) of the VPN interface. The recommended range is 1500-1900 bytes. 

#### Restore VPN automatically

If enabled, this feature automatically re-enables AdGuard’s local VPN after it has been turned off due to network absence, tethering, or low-power mode.

#### Packet capture (PCAP)

> If enabled, AdGuard will create the special file name `timestamp>.pcap` (for instance, 1682599851461.pcap). It contains all network packets transferred through the VPN. This file is located in the app cache directory and can be analyzed with the Wireshark program.

Reviewer 2: `timestamp>.pcap` -> `timestamp.pcap`

#### Include Wi-Fi gateway in VPN routes

If you want the gateway IP addresses to be added to VPN routes when on Wi-Fi, enable this feature.

#### IPv4 address

Here you can enter IP address that will be used to create a TUN interface. By default, it is `172.18.11.218`.

#### Forcibly route LAN IPv4

When enabled, AdGuard filters all LAN connections, including local IPv4 network traffic, even if the *Route all LAN IPv4 connections* option is enabled.

#### Route all LAN IPv4 connections

Once enabled, AdGuard excludes LAN connections from filtering for simple networks. May not work for complex networks. Works only with the *Forcibly route LAN IPv4* option disabled.

#### IPv6 address

> Here you can enter IP address that will be used to create a TUN interface. By default, it is `2001:db8:ad:0:ff::`.

Reviewer: ```suggestion
Here you can enter the IP address that will be used to create a TUN interface. By default, it is `2001:db8:ad:0:ff::`.
```

### Miscellaneous

#### Detect Samsung Pay

> Korean users may encounter issues with Samsung Pay when AdGuard is enabled. Activate this feature to pause AdGuards’ protection and use the Samsung Pay app seamlessly.

Reviewer: ```suggestion
Korean users may encounter issues with Samsung Pay when AdGuard is enabled. Activate this feature to pause AdGuard's protection and use the Samsung Pay app seamlessly.
```
